test/artifact_bucket_test.go

```diff
@@ -0,0 +1,243 @@
+// +build e2e
+
+/*
+Copyright 2018 Knative Authors LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1"
+	tb "github.com/knative/build-pipeline/test/builder"
+	knativetest "github.com/knative/pkg/test"
+	"github.com/knative/pkg/test/logging"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	helloworldResourceName    = "helloworldgit"
+	addFileTaskName           = "add-file-to-resource-task"
+	readFileTaskName          = "read-new-file-task"
+	bucketTestPipelineName    = "bucket-test-pipeline"
+	bucketTestPipelineRunName = "bucket-test-pipeline-run"
+	systemNamespace           = "knative-build-pipeline"
+	bucketSecretName          = "bucket-secret"
+	bucketSecretKey           = "bucket-secret-key"
+)
+
+// TestStorageBucketPipelineRun is an integration test that will verify a pipeline
+// can use a bucket for temporary storage of artifacts shared between tasks
+func TestStorageBucketPipelineRun(t *testing.T) {
+	configFilePath := os.Getenv("KANIKO_SECRET_CONFIG_FILE")
```

Comment:
quick question: do we know if `KANIKO_SECRET_CONFIG_FILE` is defined in our automated tests?

Reply:
the user that is used to run the e2e tests doesn't have high permissions, so it would be able to execute the `gcs` and `artifact_bucket` tests anyways